### PR TITLE
refactor(mito): define manifest related API

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

There are too many abstractions around manifest. They were needed when we had region manifest and table manifest. In #1869 we are going to merge these two manifests. By doing this the implementation can also be simplified. Major change:
- trait `Checkpoint` -> struct `RegionCheckpoint`
- trait `MetaAction` -> struct `RegionMetaActionList`
- trait `MetaActionIterator` -> struct `MetaActionIteratorImpl`

The next step is to port `manifest/region.rs` and `manifest/storage.rs`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by evenyag — APPROVED** [review]
Great. But where is your mermaid?

**Review by v0y4g3r — APPROVED** [review]
LGTM, todo!() master :sweat_smile:

File: src/mito2/src/manifest/action.rs [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+use serde::{Deserialize, Serialize};
+use storage::metadata::{ColumnFamilyMetadata, ColumnMetadata, VersionNumber};
+use storage::sst::{FileId, FileMeta};
+use store_api::manifest::action::{ProtocolAction, ProtocolVersion};
+use store_api::manifest::ManifestVersion;
+use store_api::storage::{RegionId, SequenceNumber};
+
+use crate::manifest::helper;
+
+/// Minimal data that could be used to persist and recover [RegionMetadata](crate::metadata::RegionMetadata).
```
Comment (evenyag) [inline src/mito2/src/manifest/action.rs:26]: non-blocking: The path to RegionMetadata is incorrect.

**Comment by waynexia** [comment]
> Great. But where is your mermaid❓

coming soon 🥲

**Pull request merged**